test/functional/feature_reindex_readonly.py

```diff
@@ -73,10 +74,10 @@ def reindex_readonly(self):
                 pass
 
         if undo_immutable:
-            self.log.info("Attempt to restart and reindex the node with the unwritable block file")
-            with self.nodes[0].assert_debug_log(expected_msgs=['FlushStateToDisk', 'failed to open file'], unexpected_msgs=[]):
-                self.nodes[0].assert_start_raises_init_error(extra_args=['-reindex', '-fastprune'],
-                    expected_msg="Error: A fatal internal error occurred, see debug.log for details")
+            self.log.debug("Attempt to restart and reindex the node with the unwritable block file")
+            with self.nodes[0].wait_for_debug_log([b"Reindexing finished"]):
```

Comment:
I'm not sure I understand. The previous code expected a crash so the test would know exactly when to check for the expected log message. In the new code bitcoind just runs fine so we have to poll the log for the success message.

Reply:
OK I see now thanks. So I guess... `start_node` calls `wait_for_rpc_connection`. Is RPC available before reindexing is finished? If we can rely on that to prevent race conditions then this could just be `assert_debug_log`.

Reply:
Got it, thanks. Worth opening a new PR?